core/src/main/scala/org/http4s/rho/bits/PathTree.scala

```diff
@@ -88,13 +88,11 @@ private[rho] object PathTree {
 
       case c @ CodecRouter(_, parser) =>
         Leaf { (req, pathstack) =>
-          RuleExecutor.runRequestRules(req, c.router.rules, pathstack).map{ i =>
-            parser.decode(req, false).run.flatMap(_.fold(e =>
-              e.toHttpResponse(req.httpVersion),
-              { body =>
-                // `asInstanceOf` to turn the untyped HList to type T
-                route.action.act(req, (body :: i).asInstanceOf[T])
-              }))
+          RuleExecutor.runRequestRules(req, c.router.rules, pathstack).flatMap{ i =>
+            parser.parse(req).run.map{ body =>
```

Comment:
Perhaps instead of my changes, we allow people to optionally pass in an error handler during the construction of a `CodecRouter`.  The error handler would then be used here in a `fold`. Basically just like the old way, but without a hard-coded failure case.